upstairs/src/lib.rs

```diff
@@ -215,6 +219,17 @@ pub trait BlockIO: Sync {
 
         self.activate().await
     }
+    /// Conditional activate if not active.
+    async fn conditional_activate_with_gen(
```

Comment:
It looks like this function is only needed by `Volume::activate_with_gen`, so I'd suggest removing it from the trait and just doing this logic in the body of that function (`volume.rs:615,626`).

Reply:
Same with `conditional_activate`, now that I look at it!

Reply:
I've removed all the `conditional_activate()` and now the behavior is:
If we are active with the same gen as requested, then return OK()
If we are not active, then activate with the provide gen.
Else, return error.